### DISCUSSION TITLE
clangd-index-server errors for system headers

### DISCUSSION THREAD

**Discussion by Badhi** [clangd/clangd#1172] [Q&A]
I'm trying out the remote indexing feature with clangd and taking the neovim build setup as the use case. (located in `/mnt/e/Work/neovim/src/neovim`)

Before running the server, I created the index using the `clangd-indexer` (located in `/mnt/e/Work/neovim/src/neovim/neovim.dex `)

The server is started with the following command

```
clangd-index-server --log=info --trace-file=index_server.log /mnt/e/Work/neovim/src/neovim/neovim.dex /mnt/e/Work/neovim/src/neovim
```

When querying references, I could see the following errors reported by the server

```
E[09:45:16.988] Unable to convert Symbol to protobuf: File path '/usr/include/x86_64-linux-gnu/sys/types.h' doesn't start with '/mnt/e/Work/neovim/src/neovim/'
```

I believe this is because the server is trying to set the URI s relative to the root of the project. But for system files, trimming the absolute path is not required. Is there a way to get this error fixed?

**Comment by Yurii-huang** [comment]
try command --path-mappings with Clangd

**Reply by Badhi** [reply]
@FunningC0217 I'm sorry could you please elaborate? How is giving this argument to `clangd` going to fix an error message on the `clangd-index-server` ?

**Reply by Yurii-huang** [reply]
> @FunningC0217 I'm sorry could you please elaborate? How is giving this argument to `clangd` going to fix an error message on the `clangd-index-server` ?

The above information is not enough to support the analysis. You should post the process of agreement request.
This is just my guess. Location the same of remote system path?
If it is inconsistent, command --path-mappings should be set when "Clangd" is started.

**Comment by ihtycm** [comment]
Does the issus has anything progress?
